core/src/trezor/wire/message_handler.py

```
@@ -35,17 +36,17 @@ def wrap_protobuf_load(
-                __name__, "received message contents:\n%s", utils.dump_protobuf(msg)
-            )
+            what = "received message contents" if is_message else "decoded protobuf"
+            log.debug(__name__, "%s:\n%s", what, utils.dump_protobuf(msg))
         return msg
     except Exception as e:
         if __debug__:
             log.exception(__name__, e)
+        what = "message" if is_message else "protobuf"
         if e.args:
-            raise DataError("Failed to decode message: " + " ".join(e.args))
+            raise DataError(f"Failed to decode {what}: " + " ".join(e.args))
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)